pkg/driver/screen/x11capture_linux.go

```diff
@@ -1,11 +1,31 @@
 package screen
 
 // #cgo pkg-config: x11 xext
+// #include <stdint.h>
 // #include <sys/shm.h>
 // #include <X11/Xlib.h>
 // #define XUTIL_DEFINE_FUNCTIONS
 // #include <X11/Xutil.h>
 // #include <X11/extensions/XShm.h>
+// void copyRGBA(void *dst, char *src, size_t l) { // 64bit aligned copy
+//   uint64_t *d = (uint64_t*)dst;
+//   uint64_t *s = (uint64_t*)src;
+//   l /= 8;
+//   for (size_t i = 0; i < l; i ++) {
+//     uint64_t v = *s;
+//     *d = 0xFF000000FF000000 |
+//          ((v >> 16) & 0xFF00000000) | (v & 0xFF0000000000) | ((v & 0xFF00000000) << 16) |
+//          ((v >> 16) & 0xFF) | (v & 0xFF00) | ((v & 0xFF) << 16);
+//     d++;
+//     s++;
+//   }
+// }
+// char *align64(char *ptr) { // return 64bit aligned pointer
+//   if (((size_t)ptr & 0xFF) == 0) {
+//     return ptr;
+//   }
+//   return (char*)(((size_t)ptr & (~(size_t)0xFF)) + 0x100);
+// }
```

Comment:
I couldn't read this... ^^, do you mind sharing what this does?

Reply:
(oops, 0xFF should be 0x07, and 0x100 should be 0x08.)
It clears lower 3 bits by `& ~0x07` (on 32bit env, it's equivalent of `& 0xFFFFFFF8`) and add 8 if there was cleared bits to avoid entering lower bound of the memory region.
So, the address gets multiple of 8bytes.

Reply:
I still don't get it.... but, oh well, it's alright. thanks

Reply:
It's basically for speeding up memory read/write.
64bit CPU can read/write 64bit data by single operation if the address is aligned. (In most case of recent environment, compiler and core library automatically align it on allocation.)